`undefined reference` error on Debian

When I run `make` on Debian 12, I get a lot of errors 

```
gcc -g -O2 -Wall -lX11 -o click-actions/raise click-actions/src/raise.c click-actions/src/common.o windowlist.o
/usr/bin/ld: /tmp/cc4tyonG.o: in function `main':
/tmp/windowlist/click-actions/src/raise.c:12: undefined reference to `XOpenDisplay'
/usr/bin/ld: /tmp/windowlist/click-actions/src/raise.c:16: undefined reference to `XMapRaised'
/usr/bin/ld: /tmp/windowlist/click-actions/src/raise.c:17: undefined reference to `XCloseDisplay'
/usr/bin/ld: click-actions/src/common.o: in function `client_msg':
/tmp/windowlist/click-actions/src/common.c:14: undefined reference to `XInternAtom'
/usr/bin/ld: /tmp/windowlist/click-actions/src/common.c:27: undefined reference to `XSendEvent'
/usr/bin/ld: windowlist.o: in function `get_property':
/tmp/windowlist/windowlist.c:32: undefined reference to `XInternAtom'
/usr/bin/ld: /tmp/windowlist/windowlist.c:50: undefined reference to `XGetWindowProperty'
/usr/bin/ld: /tmp/windowlist/windowlist.c:74: undefined reference to `XFree'
/usr/bin/ld: /tmp/windowlist/windowlist.c:57: undefined reference to `XFree'
/usr/bin/ld: windowlist.o: in function `calculate_window_middle_x_y':
/tmp/windowlist/windowlist.c:140: undefined reference to `XGetGeometry'
/usr/bin/ld: /tmp/windowlist/windowlist.c:141: undefined reference to `XTranslateCoordinates'
/usr/bin/ld: windowlist.o: in function `get_window_title':
/tmp/windowlist/windowlist.c:191: undefined reference to `XInternAtom'
/usr/bin/ld: windowlist.o: in function `generate_window_list':
/tmp/windowlist/windowlist.c:251: undefined reference to `XSetErrorHandler'
/usr/bin/ld: /tmp/windowlist/windowlist.c:267: undefined reference to `XSetErrorHandler'
collect2: error: ld returned 1 exit status
make: *** [Makefile:18: click-actions/raise] Error 1
```

--- a/Makefile
+++ b/Makefile
@@ -6,7 +6,7 @@ LDFLAGS = -lX11
 all: main click-actions/raise click-actions/minimize click-actions/close config.toml
 
 main: main.c windowlist.o windowlist.h toml-c.h
-	gcc -DCOMPILE_DIR=\"$(shell pwd)\" $(CFLAGS) $(LDFLAGS) -o main main.c windowlist.o
+	gcc -DCOMPILE_DIR=\"$(shell pwd)\" $(CFLAGS) -o main main.c windowlist.o $(LDFLAGS)
 
 windowlist.o: windowlist.c
 	gcc $(CFLAGS) -c -o $@ $^

--- a/Makefile
+++ b/Makefile
@@ -15,13 +15,13 @@ click-actions/src/common.o: click-actions/src/common.c
 	gcc $(CFLAGS) -c -o $@ $^
 
 click-actions/raise: click-actions/src/raise.c click-actions/src/common.o windowlist.o
-	gcc $(CFLAGS) $(LDFLAGS) -o $@ $^
+	gcc $(CFLAGS) -o $@ $^ $(LDFLAGS)
 
 click-actions/minimize: click-actions/src/minimize.c click-actions/src/common.o
-	gcc $(CFLAGS) $(LDFLAGS) -o $@ $^
+	gcc $(CFLAGS) -o $@ $^ $(LDFLAGS)
 
 click-actions/close: click-actions/src/close.c click-actions/src/common.o
-	gcc $(CFLAGS) $(LDFLAGS) -o $@ $^
+	gcc $(CFLAGS) -o $@ $^ $(LDFLAGS)
 
 config.toml: config-default.toml
 	@if [ ! -f config.toml ]; then \

